tutorials/tour/_posts/2017-02-13-higher-order-functions.md

```diff
@@ -9,34 +9,112 @@ categories: tour
 num: 8
 next-page: nested-functions
 previous-page: anonymous-function-syntax
+assumed-knowledge: sequence-comprehensions, anonymous-function-syntax
 ---
 
-Scala allows the definition of higher-order functions. These are functions that _take other functions as parameters_, or whose _result is a function_. Here is a function `apply` which takes another function `f` and a value `v` and applies function `f` to `v`:
-
+Higher order functions take other functions as parameters or return a function as
+a result. This is possible because functions are first-class values in Scala.
+One of the most common examples is the higher-order
+function `map` which is available for collections in Scala.
 ```tut
-def apply(f: Int => String, v: Int) = f(v)
+val salaries = Seq(20000, 70000, 40000)
+val doubleSalary = (x: Int) => x * 2
+val newSalaries = salaries.map(doubleSalary) // List(40000, 140000, 80000)
 ```
+`doubleSalary` is a function which takes a single Int, `x`, and returns `x * 2`. In general, the tuple on the left of the arrow `=>` is a parameter list and the value of the expression on the right is what gets returned. One line 3, the function `doubleSalary` gets applied to each element in the
+list of salaries.
 
-_Note: methods are automatically coerced to functions if the context requires this._
+To shrink the code, we could make the function anonymous and pass it directly as
+an argument to map:
+```
+val salaries = Seq(20000, 70000, 40000)
+val newSalaries = salaries.map(x => x * 2) // List(40000, 140000, 80000)
+```
+Notice how `x` is not declared as an Int in the above example. That's because the
+compiler can infer the type based on the type of function map expects. An even more idiomatic way to write the same piece of code would be
 
-Here is another example:
- 
 ```tut
-class Decorator(left: String, right: String) {
-  def layout[A](x: A) = left + x.toString() + right
+val salaries = Seq(20000, 70000, 40000)
+val newSalaries = salaries.map(_ * 2)
+```
+Since the Scala compiler already knows the type of the parameters (a single Int),
+ you just need to provide the right side of the function. The only
+caveat is that you need to use `_` in place of a parameter name (it was `x` in
+the previous example).
```

Comment:
Just use "difference" instead of "caveat".
If you want to make it more clear, you could also add Julien's explanation but call it a "slot" that has to be filled.